x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/datastreams/DataStreamFeatureSetUsage.java

```diff
@@ -77,26 +84,50 @@ public static class DataStreamStats implements Writeable {
 
         private final long totalDataStreamCount;
         private final long indicesBehindDataStream;
-
-        public DataStreamStats(long totalDataStreamCount, long indicesBehindDataStream) {
+        private final long failureStoreEnabledDataStreamCount;
+        private final long failureStoreIndicesCount;
+
+        public DataStreamStats(
+            long totalDataStreamCount,
+            long indicesBehindDataStream,
+            long failureStoreEnabledDataStreamCount,
+            long failureStoreIndicesCount
+        ) {
```

Comment:
If we convert this to a record, we can remove this constructor, `hashcode` and `equals`. Again, not a blocker for me if you feel it's out of scope.

Reply:
done